### PR TITLE
Fix the Body type parameters in the Rest API

### PR DESCRIPTION
**Pull request by Auriga2**
### Description of the Change
The parameters of `body` type in the REST API are broken. The request doesn't add any content for any user input.
E.g.:
plugins parameter in run_plugins endpoint
args parameter in run_plugin endpoint
data parameter in add_recordstore endpoint

### Alternate Designs
N/A

### Why Should This Be In Core?
Fixes the bug

### Benefits
The endpoints are now usable with json type parameters. This can be improved to contain `Example Value` which will be fixed in another case.

### Possible Drawbacks
N/A

### Verification Process
1. Open Swagger (Run constellation and `Tools`->`Start Rest Server` then `Display Rest Server Documentation)`
2. Try any of the above endpoints in the REST API with a valid json input
3. Verify it worked with a successful response message and the request should contain the JSON you entered.

### Applicable Issues
https://github.com/constellation-app/constellation/issues/654

### CODE REVIEW

File: CoreWebServer/src/au/gov/asd/tac/constellation/webserver/api/SwaggerServlet.java [hunk 1]
```diff
@@ -122,13 +122,24 @@ protected void doGet(final HttpServletRequest request, final HttpServletResponse
                     final ArrayNode params = httpMethod.putArray("parameters");
                     rs.createParameters().getParameters().entrySet().forEach(entry -> {
                         final PluginParameter<?> pp = entry.getValue();
+
+                        if (pp.getName().toLowerCase(Locale.US).contains("(body)")) {
```
Comment (antares1470) [inline CoreWebServer/src/au/gov/asd/tac/constellation/webserver/api/SwaggerServlet.java:126]: Are you able to change `Locale.US` to `Locale.ENGLISH`? This just keeps it consistent with the rest of the codebase

**Comment by formalhaut69** [comment]
image
I got the correct response so it appears to be working for me.

File: CoreWebServer/src/au/gov/asd/tac/constellation/webserver/api/SwaggerServlet.java (hunk 1 above)
Comment (Auriga2) [reply CoreWebServer/src/au/gov/asd/tac/constellation/webserver/api/SwaggerServlet.java:126]: @antares1470 It appears in 3 files : FontInfo, GlyphsFrame as well. Shall I change them all?

Comment (Auriga2) [reply CoreWebServer/src/au/gov/asd/tac/constellation/webserver/api/SwaggerServlet.java]: Fixed them all.

**Comment by arcturus2** [comment]
hmm sorry but I don't follow @Auriga2. i tested `master` and your branch and i got the same request header.

image

I'm not seeing any change in this branch compared to master.

File: CoreWebServer/src/au/gov/asd/tac/constellation/webserver/api/SwaggerServlet.java [hunk 2]
```diff
@@ -122,20 +124,31 @@ protected void doGet(final HttpServletRequest request, final HttpServletResponse
...
-                        param.put("in", pp.getName().toLowerCase(Locale.US).contains("(body)") ? "body" : "query");
-                        param.put("required", false); // TODO Hard-code this until PluginParameters grows a required field.
-                        param.put(DESCRIPTION, pp.getDescription());
-                        final ObjectNode schema = param.putObject(SCHEMA);
-                        schema.put("type", pp.getType().getId());
+
+                        if (pp.getName().toLowerCase(Locale.ENGLISH).contains("(body)")) {
+                            final ObjectNode requestBody = httpMethod.putObject("requestBody");
+                            requestBody.put(DESCRIPTION, pp.getName().replace("(body)", " - ") + pp.getDescription());
+                            requestBody.put(REQUIRED, false); //fix in the other ticket
```
Comment (arcturus2) [inline CoreWebServer/src/au/gov/asd/tac/constellation/webserver/api/SwaggerServlet.java:131]: @Auriga2 can you please prefix this comment with "TODO" and you can mention the GitHub issue like #123. Thanks.

**Comment by Auriga2** [comment]
> hmm sorry but I don't follow @Auriga2. i tested `master` and your branch and i got the same request header.
> 
> image
> 
> I'm not seeing any change in this branch compared to master.

@arcturus2 Is this image from master, without my changes? Did you check the master branch after this branch? If that is the case, may be it had these changes cached?
Anyway I'll double check in current master.

**Comment by Auriga2** [comment]
@arcturus2 My assumption is correct. In Master I cannot see my changes. This is the `run_plugins` endpoint. Th ePlugins parameter is different:
image

**Comment by arcturus2** [comment]
@Auriga2 oww i think your right. I had the web page opened and I don't think I hit the refresh button when I switched to yours. That makes perfect sense now - sorry about that one. I'll test this again.

File: CoreWebServer/src/au/gov/asd/tac/constellation/webserver/api/SwaggerServlet.java (hunk 2 above)
Comment (Auriga2) [reply CoreWebServer/src/au/gov/asd/tac/constellation/webserver/api/SwaggerServlet.java]: Fixed.

**Comment by arcturus2** [comment]
@Auriga2 perhaps my old age is making it difficult for me to understand the difference. I've tested with your branch again and I don't see the difference between the call made in your branch and master. Master seems to work fine for me too for those endpoints.

Are you able to explicitly show what the difference is with master vs this PR and the benefit please? Perhaps screenshots and highlighting the important bit or something like that is required.

**Comment by Auriga2** [comment]
@arcturus2 This is the difference: The `broken` screenshot is from v2.0.0-rc1
Also try opening Swagger using `Tools`->`Start Rest Server` then `Display Rest Server Documentation`
image

**Comment by aldebaran30701** [comment]
I've just tested this in the executable V2.0 rc2 and can confirm that the screenshot above, on the right hand side is what I see.
Running this branch, I see the new change which is the picture on the left. 

This I'm assuming is what @Auriga2 is talking about. @arcturus2

**Comment by arcturus2** [comment]
Thanks again @Auriga2. Testing your PR again I can clearly see what you have fixed. I was looking from developer tools in Chrome and was getting confused as to what I saw. I've realised the bug you have fixed too because its not working on `master`. Appreciate your patience here.

Thank you @aldebaran30701 for testing this and for the feedback.

**Pull request merged**